qa/rpc-tests/mining_shielded_coinbase.py

```diff
@@ -65,24 +66,18 @@ def run_test (self):
         assert_equal(self.nodes[1].z_getbalance(node1_zaddr), 0)
 
         # Stop node 1 and check logs to verify the block was rejected correctly
-        print("Checking node 1 logs")
-        self.nodes[1].stop()
-        bitcoind_processes[1].wait()
-        logpath = self.options.tmpdir + "/node1/regtest/debug.log"
-        foundErrorMsg = False
-        with open(logpath, "r") as myfile:
-            logdata = myfile.readlines()
-        for logline in logdata:
-            if "CheckTransaction(): coinbase has output descriptions" in logline:
-                foundErrorMsg = True
-                break
-        assert(foundErrorMsg)
-
-        # Restart node 1
-        self.nodes[1] = self.start_node_with(1, [
-            "-mineraddress=%s" % node1_zaddr,
-        ])
-        connect_nodes(self.nodes[1], 0)
+        string_to_find = "CheckTransaction(): coinbase has output descriptions"
+        check_node_log(self, 1, ["-mineraddress=%s" % node1_zaddr], string_to_find)
+
+        # Test assert in check_node_log
+        try:
+            string_to_find = "Will not be found"
+            check_node_log(self, 1, ["-mineraddress=%s" % node1_zaddr], string_to_find)
+        except AssertionError:
+            print("Assert OK")
+            # Restart node 1
+            self.nodes[1] = self.start_node_with(1, ["-mineraddress=%s" % node1_zaddr])
+            connect_nodes(self.nodes[1], 0)
```

Comment:
This is testing the test framework, not the code under test. Therefore it should be in a separate file, I suggest calling it `framework.py` since there isn't a suitable file already.

Also you can use [`assert_raises`](https://github.com/zcash/zcash/blob/35bff6ac7cbe3c22e8fab7dd9269b743e89129e0/qa/rpc-tests/test_framework/util.py#L430-L438) and simplify, because if it's done separately, there is no need to restart the node and no need to pass `-mineraddress`.